src/components/Icon/Icon.scss

```diff
@@ -27,192 +27,394 @@
+  }
 }
 
 .colorBlack {
   @include color-icon('black');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
+  }
 }
 
 .colorSkyLighter {
   @include color-icon('sky', 'lighter');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
+  }
 }
 
 .colorSkyLight {
   @include color-icon('sky', 'light');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
+  }
 }
 
 .colorSky {
   @include color-icon('sky');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
+  }
 }
 
 .colorSkyDark {
   @include color-icon('sky', 'dark');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
+  }
 }
 
 .colorInkLightest {
   @include color-icon('ink', 'lightest');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
+  }
 }
 
 .colorInkLighter {
   @include color-icon('ink', 'lighter');
 
   &::before {
-    background-color: color('sky');
+    background-color: var(--p-surface, color('sky'));
+  }
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
   }
 }
 
 .colorInkLight {
   @include color-icon('ink', 'light');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
+  }
 }
 
 .colorInk {
   @include color-icon('ink');
 
   &::before {
-    background-color: color('sky');
+    background-color: var(--p-surface, color('sky'));
+  }
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
   }
 }
 
 .colorBlueLighter {
   @include color-icon('blue', 'lighter');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
+  }
 }
 
 .colorBlueLight {
   @include color-icon('blue', 'light');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
+  }
 }
 
 .colorBlue {
   @include color-icon('blue');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
+  }
 }
 
 .colorBlueDark {
   @include color-icon('blue', 'dark');
 
-  &::before {
-    background-color: color('blue', 'light');
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
   }
 }
 
 .colorBlueDarker {
   @include color-icon('blue', 'darker');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
+  }
 }
 
 .colorIndigoLighter {
   @include color-icon('indigo', 'lighter');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-action-primary));
+  }
 }
 
 .colorIndigoLight {
   @include color-icon('indigo', 'light');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-action-primary));
+  }
 }
 
 .colorIndigo {
   @include color-icon('indigo');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-action-primary));
+  }
 }
 
 .colorIndigoDark {
   @include color-icon('indigo', 'dark');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-action-primary));
+  }
 }
 
 .colorIndigoDarker {
   @include color-icon('indigo', 'darker');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-action-primary));
+  }
 }
```

Comment:
I've made a blanket decision to alias all indigo icons to `primary`, since this is our branded variant.

Reply:
Sounds good to me! Maybe another thing to flag for the design review to note in the polaris-ux issue?